src/LightDFG/services/LightDFG.jl

```diff
@@ -391,7 +391,7 @@ Optionally provide a distance to specify the number of edges should be followed.
 Optionally provide an existing subgraph addToDFG, the extracted nodes will be copied into this graph. By default a new subgraph will be created.
 Note: By default orphaned factors (where the subgraph does not contain all the related variables) are not returned. Set includeOrphanFactors to return the orphans irrespective of whether the subgraph contains all the variables.
 """
-function getSubgraphAroundNode(dfg::LightDFG{P,V,F}, node::DFGNode, distance::Int64=1, includeOrphanFactors::Bool=false, addToDFG::LightDFG=LightDFG{P,V,F}(); solvable::Int=0)::LightDFG where {P <: AbstractParams, V <: AbstractDFGVariable, F <: AbstractDFGFactor}
+function getSubgraphAroundNode(dfg::LightDFG{P,V,F}, node::DFGNode, distance::Int64=1, includeOrphanFactors::Bool=false, addToDFG::G=LightDFG{P,V,F}(); solvable::Int=0)::AbstractDFG where {P <: AbstractParams, V <: AbstractDFGVariable, F <: AbstractDFGFactor, G <: AbstractDFG}
```

Comment:
still curious why we have a specialized function here, must remove in time

Reply:
On philosophy, its about "cross-feeding" data from one DFG to another DFG and any whicha-way and abstracting away some of the technology concerns -- if its slightly slower, thats fine.  If it's much much slower with absolutely no recourse via a common abstract definition, then we can look at a performance specialization for high throughput cases.  See similar request there too: https://github.com/JuliaRobotics/DistributedFactorGraphs.jl/issues/214#issuecomment-558186644

Also separating out the design discussion:
#264